AGENTGATEWAY.md

```diff
@@ -0,0 +1,30 @@
+# Agentgateway Support in Istio
+
+Istio has **experimental** support for running the [agentgateway](https://github.com/agentgateway/agentgateway/) proxy as a dataplane. Agentgateway is a high-performance, lightweight proxy designed for cloud-native AI applications. Its primary control plane is the [kgateway](https://github.com/kgateway-dev/kgateway/) CNCF sandbox project, and the two together are fully conformant with the Gateway API. This document outlines the contract between the Istio and agentgateway projects and sets explicit milestones and advancement criteria for continued integration.
+
+## Scope of the Integration
+
+Agentgateway has a variety of features that kgateway (its primary control plane) exposes through numerous custom resources. However, Istio's integration with agentgateway will expose these features __almost exclusively through Gateway API__ and related subprojects and working groups (e.g. [kube-agentic-networking](https://github.com/kubernetes-sigs/kube-agentic-networking) and [wg-ai-gateway](https://github.com/kubernetes-sigs/wg-ai-gateway/)). This means that Istio users will be able to configure and manage agentgateway proxies using Gateway API resources, while other custom resources specific to kgateway will not be supported in this integration. This is a purposeful decision to maintain a clear and focused integration between Istio and agentgateway, leveraging the standardized Gateway API for configuration and management. The Istio TOC can approve exceptions for specific resources on a case-by-case basis if there is a compelling use case and no viable alternative using Gateway API.
```

Comment:
Seems no api is designed yet from the posted link

Reply:
"The Istio TOC can approve exceptions for specific resources on a case-by-case basis if there is a compelling use case and no viable alternative using Gateway API."  -  What are some of examples for this and What would it look like? Do we import kgateway custom resources to Istio or build equivalent in Istio using Istio APIs? I would lean towards building a custom Istio API for such use cases - but they have to be pretty exception cases. Alternatively can we enhance agentgateway with some type of extension similar to what we do with Envoy filters that people can use?